crates/merkle-tree/src/class.rs

```diff
@@ -71,6 +74,162 @@ impl<'tx> ClassCommitmentTree<'tx> {
         let commitment = ClassCommitment(update.root_commitment);
         Ok((commitment, update))
     }
+
+    /// Generates a proof for a given `key`
+    pub fn get_proof(
+        tx: &'tx Transaction<'tx>,
+        block: BlockNumber,
+        class_hash: ClassHash,
+    ) -> anyhow::Result<Option<Vec<TrieNode>>> {
+        let root = tx
+            .class_root_index(block)
+            .context("Querying class root index")?;
+
+        let Some(root) = root else {
+            return Ok(None);
+        };
+
+        let storage = ClassTrieStorage {
+            tx,
+            block: Some(block),
+        };
+
+        let casm = tx
+            .casm_hash_at(block.into(), class_hash)
+            .context("Querying CASM hash")?;
+
+        let Some(casm) = casm else {
+            return Ok(None);
+        };
+
+        MerkleTree::<PedersenHash, 251>::get_proof(root, &storage, casm.view_bits())
```

Comment:
The class trie uses Poseidon hashing. I'm unsure of how much the hash function generic is useful in the function you call here but I'm pretty sure we should use `PoseidonHash` for correctness.